labrad/decorators.py

```diff
@@ -21,202 +21,140 @@
 
 from __future__ import absolute_import
 
-from functools import wraps
-from inspect import getargspec
+import functools
+import inspect
+import itertools
 
-from twisted.internet.defer import inlineCallbacks
+import twisted.internet.defer as defer
 
 from labrad import types as T, util
 
-def _isGenerator(f):
-    """Check to see whether f is a generator.
-
-    See the documentation on code objects at:
-    http://docs.python.org/ref/types.html
-    """
-    return bool(f.func_code.co_flags & 0x20)
-
 def _product(lists):
     """Return the cartesian product of a list of lists."""
     if not len(lists): return [[]]
     return [[h] + t for h in lists[0] for t in _product(lists[1:])]
 
-class Setting(object):
-    def __init__(self, func):
-        self.func = func
+def setting(lr_ID, lr_name=None, returns=[], unflatten=True, **params):
```

Comment:
This is the method people are going to call, rather than the `Setting` class, so please add a doc string, even if just to point to the docstring on the `Setting` class.